Some housekeeping for 2019 - 2022.

marcfehling opened the pull request:
- Fix `month` entries (see Q13 of [bibtex FAQ](https://ctan.org/pkg/bibtex?lang=en))
- Fix `doi` entries (no urls)
- Remove `url` if information redundant to `doi` field
- Update preprints if published in the meantime. Let `url` still point to arxiv preprint.

There are tons of additional `arxiv` preprints for previous years through which we need to go. Part of #289.

bangerth commented on publications-2020.bib:
```diff
@@ -124,7 +123,7 @@ @Article{WiWo20
 keywords = {gradient-based optimization, monolithic formulation, optimal control, optimal design, unsteady nonlinear fluid-structure interaction},
 doi = {10.1002/nme.6372},
-url = {https://onlinelibrary.wiley.com/doi/abs/10.1002/nme.6372},
+url = {https://arxiv.org/abs/1910.03424},
```
Same issue here

bangerth commented on publications-2020.bib:
```diff
@@ -157,8 +148,8 @@ @Article{EndtNeiLaWiWo20
 issn = "0898-1221",
-doi = "https://doi.org/10.1016/j.camwa.2020.01.005",
-url = "http://www.sciencedirect.com/science/article/pii/S0898122120300122",
+doi = "10.1016/j.camwa.2020.01.005",
+url = "https://arxiv.org/abs/1903.02799",
```
And here

bangerth commented on publications-2020.bib:
```diff
@@ -933,10 +876,9 @@ @article{BorregalesRevern2020
 @article{McBride2020,
   doi = {10.1016/j.cma.2020.113320},
-  url = {https://doi.org/10.1016/j.cma.2020.113320},
+  url = {https://arxiv.org/abs/1909.08695},
```
and here

bangerth commented on publications-2021.bib:
```diff
@@ -470,13 +446,12 @@ @Article{Araujo2021b
-  doi = {https://doi.org/10.1016/j.jcp.2020.110024},
-  url = {https://www.sciencedirect.com/science/article/pii/S0021999120307981}
+  doi = {10.1016/j.jcp.2020.110024},
+  url = {https://arxiv.org/abs/1904.08812}
```
And here

bangerth commented on publications-2021.bib:
```diff
@@ -671,13 +652,24 @@ @phdthesis{fehn2021phd
 }
 
+@article{fehn2021anomalous,
+  doi = {10.1017/jfm.2021.1003},
+  url = {https://arxiv.org/abs/2007.01656},
```
and here

bangerth commented on publications-2021.bib:
```diff
@@ -921,7 +909,7 @@ @article{heltai2021geometry
   volume = {47},
   year = {2021},
-  url = {https://dx.doi.org/10.1145/3468428},
+  url = {https://arxiv.org/abs/1910.09824},
```
and here

bangerth commented on publications-2021.bib:
```diff
@@ -992,14 +976,62 @@ @article{Comeau2021
 }
 
+@article{Bonito2021,
+  doi = {10.1142/s0218202521500044},
+  url = {https://arxiv.org/abs/1912.03812},
+  year = {2021},
+  publisher = {World Scientific Pub Co Pte Lt},
+  volume = {31},
+  number = {01},
+  pages = {133--175},
+  author = {Andrea Bonito and Ricardo H. Nochetto and Dimitrios Ntogkas},
+  title = {{DG} approach to large bending plate deformations with isometry constraint},
+  journal = {Mathematical Models and Methods in Applied Sciences}
```
here too, the URL can't be right

bangerth commented on publications-2021.bib:
```diff
@@ -992,14 +976,62 @@ @article{Comeau2021
+  journal = {Mathematical Models and Methods in Applied Sciences}
+}
+
+@incollection{Klein2021,
+  doi = {10.1007/978-3-030-57784-1_6},
+  url = {https://arxiv.org/abs/1911.04855},
```
same here

bangerth commented on publications-2021.bib:
```diff
@@ -992,14 +976,62 @@ @article{Comeau2021
+  booktitle = {Time-dependent Problems in Imaging and Parameter Identification}
+}
+
+@article{Liu2021,
+  doi = {10.1016/j.cam.2020.113219},
+  url = {https://arxiv.org/abs/1912.08004},
```
and here

bangerth commented on publications-2021.bib:
```diff
@@ -992,14 +976,62 @@ @article{Comeau2021
+  journal = {Journal of Computational and Applied Mathematics}
+}
+
+@article{Anselmann2021,
+  doi = {10.1016/j.matcom.2020.10.027},
+  url = {https://arxiv.org/abs/1912.07426},
```
and here

bangerth commented on publications-2022.bib:
```diff
@@ -591,7 +603,18 @@ @MastersThesis{Wik1676116
+  month  = jun
+}
+
+@incollection{BaMaWaWiWo22,
+  doi = {10.1007/978-3-030-92672-4_8},
+  url = {https://arxiv.org/abs/2006.16566},
```
and here

bangerth reviewed (commented):
This wasn't the most exciting patch to sift through :-) But I did, all the way to the bottom, and there are two notes I have:

* I marked up a number of places where you provide a URL for an article where the article has appeared in a journal but the URL points to arXiv. This isn't right. The preprint on arXiv is not guaranteed to be identical to the one posted on the journal's web site, and only the latter is normative. In those places, please remove the URL. The DOI is enough to click through to the article, there is no need for a separate URL -- but in particular, if we provide one, it needs to go to the article, and not a preprint.
* I'm of mixed opinion about the removal of arXiv preprint links if a similar article exists. They are often, but not always identical. For example, I have one paper that did not get accepted as originally written, so we put it onto arXiv. Later, we shrunk the article by half and got it published elsewhere. In cases such as this, it might make sense to keep both on the list. Of course, it is not practically possible (=it is not a useful use of anyone's time) to sort out when an arXiv preprint is or isn't identical to a paper. As a consequence, I don't know what to do -- but I might suggest that it isn't the best use of your time to go through all of the arXiv preprints and find out whether there is a corresponding journal article elsewhere in our list.

marcfehling commented:
> I marked up a number of places where you provide a URL for an article where the article has appeared in a journal but the URL points to arXiv. This isn't right. The preprint on arXiv is not guaranteed to be identical to the one posted on the journal's web site, and only the latter is normative. In those places, please remove the URL. The DOI is enough to click through to the article, there is no need for a separate URL -- but in particular, if we provide one, it needs to go to the article, and not a preprint.

I removed the URLs to the preprints if the bibtex entry refers to the published manuscript. We had multiple URLs pointing to preprints before this patch, so I thought this was common practice in the publication list. I removed those entries as well to have a coherent list.

> I'm of mixed opinion about the removal of arXiv preprint links if a similar article exists. They are often, but not always identical. For example, I have one paper that did not get accepted as originally written, so we put it onto arXiv. Later, we shrunk the article by half and got it published elsewhere. In cases such as this, it might make sense to keep both on the list. Of course, it is not practically possible (=it is not a useful use of anyone's time) to sort out when an arXiv preprint is or isn't identical to a paper. As a consequence, I don't know what to do -- but I might suggest that it isn't the best use of your time to go through all of the arXiv preprints and find out whether there is a corresponding journal article elsewhere in our list.

I see your point. The case you describe seems to me to be rather the exception. In this patch, I changed arxiv preprints to published manuscripts if I could verify that the authors and the title were similar -- I did not check the contents of these articles in detail.

I like to think of preprints as early-stage research that has been advanced through peer review. Thus personally I consider preprints as duplicate entries, which should be left out if a published manuscript is available, or which should only appear alongside it. Because of this and for cases like the one you described, I liked the idea to connect the published article to the preprints (as we already did for some entries).

What do you think of _adding a separate field_ that contains links to preprints? Since we are using `biber` it shouldn't be too difficult to add such a field. And we can adjust the jabref templates to display hyperlinks as `preprint`.

As a side note: I closed #344 as one of the authors @elauksap confirmed in https://github.com/dealii/publication-list/pull/380#discussion_r960295748 that both preprints indeed became the published article.

bangerth commented:
On 9/1/22 14:13, Marc Fehling wrote:
> What do you think of /adding a separate field/ that contains links to 
> preprints? Since we are using |biber| it shouldn't be too difficult to add
>  such a field. And we can adjust the jabref templates to display hyperlinks
> as |preprint|.

Just because it is possible doesn't mean it's a good use of anyone's time :-) 
Let the following be a guiding question: Does it make the lives of a 
sufficiently large number of people better if we/you spend time on adding 
these links? I think the answer is probably no. There are better things to do :-)
